packages/epics/src/agreements/components/agreement-form-payment-schedule.tsx

```diff
@@ -0,0 +1,257 @@
+'use client';
+
+import { useEffect } from 'react';
+import {
+  useForm,
+  Controller,
+  useFieldArray,
+  FormProvider,
+} from 'react-hook-form';
+import {
+  DropdownMenu,
+  DropdownMenuTrigger,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  Button,
+  Label,
+  Input,
+  DatePicker,
+} from '@hypha-platform/ui';
+import { cn } from '@hypha-platform/lib/utils';
+import { ChevronDownIcon, PlusIcon, Cross2Icon } from '@radix-ui/react-icons';
+import { PercentIcon } from 'lucide-react';
+import { isBefore } from 'date-fns';
+
+const options = ['Immediately', 'Future Payment', 'Milestones'] as const;
+type Option = (typeof options)[number];
+
+type DateRange = {
+  from: Date | undefined;
+  to: Date | undefined;
+};
+
+type Milestone = {
+  percentage: string;
+  dateRange: DateRange;
+};
+
+type FormValues = {
+  option: Option;
+  futureDate?: Date;
+  milestones: Milestone[];
+};
+
+type AgreementFormPaymentScheduleProps = {
+  onChange?: (value: FormValues) => void;
+};
+
+export function AgreementFormPaymentSchedule({
+  onChange,
+}: AgreementFormPaymentScheduleProps) {
+  const methods = useForm<FormValues>({
+    defaultValues: {
+      option: 'Immediately',
+      futureDate: undefined,
+      milestones: [],
+    },
+    mode: 'onChange',
+  });
+
+  const {
+    control,
+    watch,
+    setValue,
+    formState: { errors },
+  } = methods;
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: 'milestones',
+  });
+
+  const selectedOption = watch('option');
+  const milestones = watch('milestones');
+  const futureDate = watch('futureDate');
+
+  useEffect(() => {
+    onChange?.({
+      option: selectedOption,
+      futureDate,
+      milestones,
+    });
+  }, [selectedOption, futureDate, milestones]);
+
+  const validateMilestones = () => {
+    const total = milestones.reduce(
+      (sum, m) => sum + Number(m.percentage || 0),
+      0,
+    );
+
+    const now = new Date();
+
+    for (let i = 0; i < milestones.length; i++) {
+      const { dateRange } = milestones[i];
+
+      if (!dateRange?.from) return 'Each milestone must have a start date';
+      if (i === 0 && isBefore(dateRange.from, now)) {
+        return 'First milestone must be in the future';
+      }
+      if (
+        i > 0 &&
+        isBefore(dateRange.from!, milestones[i - 1].dateRange.from!)
+      ) {
+        return `Milestone ${i + 1} must be after milestone ${i}`;
+      }
+    }
+
+    if (total > 100) return 'Total percentage cannot exceed 100%';
+    return true;
+  };
+
+  const validateFutureDate = (date: Date | undefined) => {
+    const now = new Date();
+    if (date && isBefore(date, now)) {
+      return 'The future payment date must be later than the current date';
+    }
+    return true;
+  };
```

Comment:
We've discussed this before. Why are those validation helper declared inline? Please keep those outside the component function body 🙏
Additionally I'm wondering why we're not utilising `zod` for the validation